### PR TITLE
Feed printer (rss/atom) to parse media files

### PR DESCRIPTION
**Pull request by mwjames**
Based on the question raised by [0], it is to allow for a feed to display things like:

image

[0] http://wikimedia.7.x6.nabble.com/display-specific-datas-in-RSS-feed-td5053425.html

### CODE REVIEW

**Comment by mwjames** [comment]
I don't have any tests for this so this needs manual testing.

File: includes/queryprinters/FeedResultPrinter.php [hunk 1]
```diff
@@ -205,15 +205,17 @@ protected function feedItem( array $row ) {
 			// Loop over all values for the property.
 			while ( ( $dataValue = $field->getNextDataValue() ) !== false ) {
 				if ( $dataValue->getDataItem() instanceof SMWDIWikipage ) {
-					$itemSegments[] = Sanitizer::decodeCharReferences( $dataValue->getLongHTMLText() );
+					$itemSegments[] = Sanitizer::decodeCharReferences(
+						$dataValue->getLongWikiText( $this->params['link'] !== 'none' ? smwfGetLinker() : null )
+					);
```
Comment (JeroenDeDauw) [inline includes/queryprinters/FeedResultPrinter.php]: You could drop the negation

File: includes/queryprinters/FeedResultPrinter.php [hunk 2]
```diff
@@ -245,31 +247,36 @@ protected function feedItem( array $row ) {
...
-			$parserOptions = new ParserOptions();
-			$parserOptions->setEditSection( false );
-
-			if ( $this->params['page'] === 'abstract' ) {
-				// Abstract of the first 30 words
-				// preg_match( '/^([^.!?\s]*[\.!?\s]+){0,30}/', $wikiPage->getText(), $abstract );
-				// $text = $abstract[0] . ' ...';
+		if ( !in_array( $this->params['page'], array( 'abstract', 'full' ) ) ) {
+			return '';
+		}
```
Comment (JeroenDeDauw) [inline includes/queryprinters/FeedResultPrinter.php]: Yay for guard clause

**Comment by JeroenDeDauw** [comment]
Ping @SemanticMediaWiki/testers

**Comment by JeroenDeDauw** [comment]
Did a quick review and looks good. Then again, I'm not even sure which part is required for the media file parsing

**Comment by kghbln** [comment]
@mwjames Is this one dead? Looks like a nice enhancement.

**Comment by mwjames** [comment]
> @mwjames Is this one dead? Looks like a nice enhancement.

Mmm ... I think there was an edge case but I cannot recall what exactly was the issue. Pushing this to the end of the 3.0 timeline.

**Comment by kghbln** [comment]
> Mmm ... I think there was an edge case but I cannot recall what exactly was the issue. Pushing this to the end of the 3.0 timeline.

No worries. To me it looked like an interesting enhancement without the need to do too much voodoo. If there is not enough time we can still move this to 3.1

**Comment by mwjames** [comment]
> No worries. To me it looked like an interesting enhancement without the need to do too much voodoo. If there is not enough time we can still move this to 3.1
> was an edge case but I cannot recall 

I still cannot remember the edge case but waiting on a divine intervention will only increase the likelihood of code rot so instead `s-0027.json` adds an integration test and the PR now becomes part of 3.0

**Comment by kghbln** [comment]
> I still cannot remember the edge case

I am sure we will find out sooner than later since this is a supposedly more heavily used feature. Even better if not at all.

**Comment by kghbln** [comment]
[Documented](https://www.semantic-mediawiki.org/w/index.php?title=Help%3AFeed_format&type=revision&diff=66531&oldid=66517)
